### PR TITLE
refactor(components/core): replace calls to ComponentFactoryResolver

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2315230](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2315230) [AB#2315232](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2315232) [AB#2315231](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2315231)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/685/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/ag-grid-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/popovers-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/tabs-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/685/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/685/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/685/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/685/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/685/playground/)

File: libs/components/core/src/lib/modules/overlay/overlay.component.ts [hunk 1]
```diff
@@ -176,17 +176,16 @@ export class SkyOverlayComponent implements OnInit, OnDestroy {
...
       providers,
       parent: this.#injector,
     });
 
-    const componentRef = this.targetRef.createComponent<C>(
-      factory,
-      undefined,
-      injector
-    );
+    const componentRef = createComponent<C>(component, {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/overlay/overlay.component.ts]: Any reason we can't use the `targetRef`'s `createComponent` method instead? Seems like a simpler implementation:
```
const componentRef = this.targetRef.createComponent(component, {
  injector,
});
```

File: libs/components/core/src/lib/modules/dock/dock.component.ts [hunk 2]
```diff
@@ -74,17 +74,16 @@ export class SkyDockComponent {
...
-    const componentRef = this.target.createComponent<T>(
-      factory,
-      undefined,
-      injector
-    );
+    const componentRef = createComponent<T>(component, {
+      environmentInjector: this.#applicationRef.injector,
+      elementInjector: injector,
+    });
+    this.target.insert(componentRef.hostView);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/dock/dock.component.ts]: Any reason we can't use the `target`'s `createComponent` method instead? Seems like a simpler implementation:
```
const componentRef = this.target.createComponent(component, {
  injector,
});
```

File: libs/components/core/src/lib/modules/dynamic-component/dynamic-component.service.ts [hunk 3]
```diff
@@ -73,9 +67,10 @@ export class SkyDynamicComponentService {
       parent: this.#injector,
     });
 
-    const componentRef = this.#componentFactoryResolver
-      .resolveComponentFactory<T>(componentType)
-      .create(injector);
+    const componentRef = createComponent<T>(componentType, {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/dynamic-component/dynamic-component.service.ts:70]: This is neat; haven't seen it before and I think it works well in the context of the dynamic component service. 👍

File: libs/components/core/src/lib/modules/overlay/overlay.component.ts (hunk 1 above)
Comment (johnhwhite) [reply libs/components/core/src/lib/modules/overlay/overlay.component.ts]: Good call. That's easier.

File: libs/components/core/src/lib/modules/dock/dock.component.ts (hunk 2 above)
Comment (johnhwhite) [reply libs/components/core/src/lib/modules/dock/dock.component.ts]: Good call. That's easier.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**